Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```
@@ -0,0 +1,382 @@
+                defaultSymbol: startingPointSymbol
+            )
+            let overlay = GraphicsOverlay()
+            overlay.renderer = renderer
+            return overlay
+        }()
+        
+        init() {
+            map.addUtilityNetwork(utilityNetwork)
+        }
+        
+        deinit {
+            ArcGISEnvironment.authenticationManager.arcGISCredentialStore.removeAll()
+        }
+        
+        /// Performs important tasks including adding credentials, loading and adding operational layers.
+        func setup() async throws {
+            try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
+            try await loadServiceGeodatabase()
+            try await loadUtilityNetwork()
+        }
+        
+        /// Load the service geodatabase and initialize the layers.
+        @MainActor
+        private func loadServiceGeodatabase() async throws {
+            // Loads the geodatabase if it does not exist.
+            if serviceGeodatabase == nil {
+                serviceGeodatabase = ServiceGeodatabase(url: .featureServiceURL)
+                try await serviceGeodatabase.load()
+                tracingActivity = .loadingNetwork
+            }
+            
+            // The gas device layer and gas line layer are created from the service geodatabase.
+            if let gasDeviceLayerTable = serviceGeodatabase.table(withLayerID: 0),
+               let gasLineLayerTable = serviceGeodatabase.table(withLayerID: 3) {
+                let layers = [gasLineLayerTable, gasDeviceLayerTable].map(FeatureLayer.init)
+                // Add the utility network feature layers to the map for display.
+                map.addOperationalLayers(layers)
+                self.layers = layers
+            }
+        }
+        
+        /// Load the utility network.
+        @MainActor
+        private func loadUtilityNetwork() async throws {
+            try? await utilityNetwork.load()
```

if load throws then the error is eaten and not propagated.  Then the status text is set to "Utility network loaded..." This probably should be `try await ...`